subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/ChildMap.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.snapshot;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+public interface ChildMap<T> {
+
+    boolean isEmpty();
+
+    List<T> values();
+
+    List<Entry<T>> entries();
+
+    void visitChildren(BiConsumer<String, ? super T> visitor);
+
+    <RESULT> RESULT getNode(VfsRelativePath targetPath, CaseSensitivity caseSensitivity, GetNodeHandler<T, RESULT> handler);
+
+    interface GetNodeHandler<T, RESULT> {
+        RESULT handleAsDescendantOfChild(VfsRelativePath pathInChild, T child);
+        RESULT handleExactMatchWithChild(T child);
+        RESULT handleUnrelatedToAnyChild();
+        RESULT handleAsAncestorOfChild(String childPath, T child);
+    }
```

Comment:
And this should not be `GetNodeHandler` but simply `NodeHandler`.

Reply:
`NodeHandler` it is.